stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -449,6 +488,268 @@ impl NakamotoChainState {
             })
     }
 
+    /// Extract and parse a nakamoto block from the DB, and verify its integrity.
+    fn load_nakamoto_block(
+        staging_db_conn: &Connection,
+        consensus_hash: &ConsensusHash,
+        block_hash: &BlockHeaderHash,
+    ) -> Result<Option<NakamotoBlock>, ChainstateError> {
+        let query = "SELECT data FROM nakamoto_staging_blocks WHERE consensus_hash = ?1 AND block_hash = ?2";
+        staging_db_conn
+            .query_row_and_then(
+                query,
+                rusqlite::params![consensus_hash, block_hash],
+                |row| {
+                    let data: Vec<u8> = row.get("data")?;
+                    let block = NakamotoBlock::consensus_deserialize(&mut data.as_slice())?;
+                    if &block.header.block_hash() != block_hash {
+                        panic!(
+                            "Staging DB corruption: expected {}, got {}",
+                            &block_hash,
+                            &block.header.block_hash()
+                        );
```

Comment:
I'm not sure how this is recoverable in Nakamoto?

Reply:
I suppose its fine to `panic!` here, but this scenario seems basically equivalent to DBErrors like `parse` errors (i.e., if `data` isn't a `Vec<u8>` or deserialize to a block, why does that error rather that `panic`?) where we typically opt to yield an error rather than immediately panicking.

Reply:
Changed to DBError